plugins/inventory/insights.py

```diff
@@ -253,16 +253,37 @@ def create_requests_authentication(self):
             self.ensure_authentication_option('client_scopes')
             return BearerAuth(self.get_oauth_token())
 
+    def ensure_requests_authentication(self):
+        """
+        Ensure the authentication object for requests is created and
+        valid/usable.
+        """
+        create_new = False
+        if not self.session.auth:
+            create_new = True
+        if create_new:
+            self.session.auth = self.create_requests_authentication()
```

Comment:
Yes: a step on top of this will be recreating the bearer authentication in case the token expired. The code for that is small, and there will be an extra check in this function.